### PR TITLE
[fix]:[CCM-11277]: Enabling the DataCheckBigQueryAndTimescale for clickhouse

### PR DESCRIPTION
**Pull request by abhinavsingh-ccm**
## Describe your changes
For clickhouse onPrem setup, we are disabling the data check job for now.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-ALL: `trigger tiAll`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43998)

### CODE REVIEW

**Review by trunapushpa — APPROVED** [review]

**Review by trunapushpa — DISMISSED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43998)* status: :shipit: complete! ✓

File: batch-processing/service/src/main/java/io/harness/batch/processing/tasklet/DataCheckBigqueryAndTimescaleTasklet.java [hunk 1]
```diff
@@ -28,9 +30,14 @@
 public class DataCheckBigqueryAndTimescaleTasklet implements Tasklet {
   @Autowired private BillingDataServiceImpl billingDataService;
   @Autowired private BigQueryHelperServiceImpl bigQueryHelperService;
+  @Autowired private BatchMainConfig config;
 
   @Override
   public RepeatStatus execute(StepContribution stepContribution, ChunkContext chunkContext) throws Exception {
+    if (DeployMode.isOnPrem(config.getDeployMode().name()) && config.isClickHouseEnabled()) {
```
Comment (sdsingh07) [inline batch-processing/service/src/main/java/io/harness/batch/processing/tasklet/DataCheckBigqueryAndTimescaleTasklet.java:37]: Let's add this check to the `EventJobScheduler#runJob` function. We can create a set and add BatchJobTypes that we don't want to run on-premises.
Then, we don't need to add this check in separate tasklets.

Comment (abhinavsingh-ccm) [reply batch-processing/service/src/main/java/io/harness/batch/processing/tasklet/DataCheckBigqueryAndTimescaleTasklet.java:42]: Didn't disable the job. Enabled the batch job for clickhouse too.

**Review by trunapushpa — APPROVED** [review]

**Review by trunapushpa — APPROVED** [review]
> Reviewed 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43998)* status: 1. unresolved discussions.

File: batch-processing/service/src/main/java/io/harness/batch/processing/billing/timeseries/service/impl/ClickHouseClusterDataService.java [hunk 2]
```diff
@@ -20,12 +23,12 @@
 import com.google.gson.JsonParser;
 import com.google.inject.Singleton;
 import java.math.BigDecimal;
-import java.sql.Connection;
-import java.sql.PreparedStatement;
-import java.sql.SQLException;
+import java.sql.*;
```
Comment (sdsingh07) [inline batch-processing/service/src/main/java/io/harness/batch/processing/billing/timeseries/service/impl/ClickHouseClusterDataService.java:26]: Need to remove this wildcard import.

**Review by sdsingh07 — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43998#-:-NOiNRmB6PPPaIN94jws:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**